data-pipeline/src/pipeline/pre_processing.py

```diff
@@ -256,22 +256,150 @@ def prepare_ks4_data(ks4_path):
     return ks4.set_index("URN")
 
 
-def prepare_central_services_data(cs_path):
-    central_services_financial = pd.read_excel(
+def prepare_central_services_data(cs_path, current_year: int):
+    central_services_financial = pd.read_csv(
         cs_path,
-        sheet_name="CentralServices",
-        usecols=input_schemas.aar_central_services.keys(),
+        usecols=lambda x: x in input_schemas.aar_central_services.keys(),
+        dtype=input_schemas.aar_central_services,
+    )
+
+    if (current_year < 2023) or (
+        "BNCH11123-BTI011-A (MAT Central services - Income)"
+        not in central_services_financial.columns
+    ):
+        central_services_financial[
+            "BNCH11123-BTI011-A (MAT Central services - Income)"
+        ] = 0.0
+
+        if (current_year <= 2022) and (
+            "BNCHBAI061 (Coronavirus Govt Funding)"
+            in central_services_financial.columns
+        ):
+            central_services_financial[
+                "BNCH11123-BTI011-A (MAT Central services - Income)"
+            ] = central_services_financial["BNCHBAI061 (Coronavirus Govt Funding)"]
+
```

Comment:
I get what is happening here, but as discussed, these are not actually substitutions but completely separate.

Prior to 2023 COVID funding was considered, so it should be accounted for.


Then, and what I think is completely separate, BNCH11123-BTI011-A was added to the raw data, and added into the mappings. The two are not mutually interchangeable and I am concerned that anyone picking up the code at a later date may think they are if we do not put a clear difference in between the two.

Reply:
What should be the logic? Is this fundamentally correct?